### PR TITLE
add policy constraints class

### PR DESCRIPTION
**Pull request by reaperhulk**
refs #1947

### CODE REVIEW

**Comment by reaperhulk** [comment]
retest this please

File: docs/x509.rst [hunk 1]
```diff
@@ -764,6 +764,40 @@ X.509 Extensions
...
+    in two ways.  It can be used to prohibit policy mapping or require
+    that each certificate in a path contain an acceptable policy
+    identifier. For more information about the use of this extension see
+    :rfc:`5280`.
+
+    .. attribute:: require_explicit_policy
+
+        :type: int or None
+
+        If this field is present, the value indicates the number of additional
```
Comment (alex) [inline docs/x509.rst]: Present is ambigious, how about "is not None"

File: docs/x509.rst [hunk 2]
```diff
@@ -764,6 +764,40 @@ X.509 Extensions
...
+        certificates that may appear in the path before an explicit policy is
+        required for the entire path. When an explicit policy is required, it
+        is necessary for all certificates in the path to contain an acceptable
+        policy identifier in the certificate policies extension.  An
+        acceptable policy identifier is the identifier of a policy required
+        by the user of the certification path or the identifier of a policy
+        that has been declared equivalent through policy mapping.
```
Comment (alex) [inline docs/x509.rst]: I've read these two sentences a dozen times, and I can't understnad what they mean. Can we dumb down the language a bit?

File: docs/x509.rst [hunk 3]
```diff
@@ -764,6 +764,40 @@ X.509 Extensions
...
+
+    .. attribute:: inhibit_policy_mapping
+
+        :type: int or None
+
+        If this field is present, the value indicates the number of additional
```
Comment (alex) [inline docs/x509.rst]: same gripe about "present"

File: docs/x509.rst [hunk 4]
```diff
@@ -764,6 +764,40 @@ X.509 Extensions
...
+.. class:: PolicyConstraints
+
+    .. versionadded:: 1.0
+
+    The policy constraints extension can be used in certificates issued
+    to CAs.  The policy constraints extension constrains path validation
+    in two ways.  It can be used to prohibit policy mapping or require
+    that each certificate in a path contain an acceptable policy
+    identifier. For more information about the use of this extension see
+    :rfc:`5280`.
```
Comment (alex) [inline docs/x509.rst]: This should mention that at least one of requires_expliti policy and inhibit policy mapping will be non-None

**Comment by reaperhulk** [comment]
We've decided to not implement this until such time as we get a user request.

**Pull request closed without merging**